### PR TITLE
When statistical information is missing, the query plan of the left deep tree is generated

### PR DESCRIPTION
**Pull request by HangyuanLiu**
#221

### CODE REVIEW

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java [hunk 1]
```diff
@@ -130,6 +131,10 @@ public OptExpression optimize(ConnectContext connectContext,
         if (!connectContext.getSessionVariable().isDisableJoinReorder()) {
             if (Utils.countInnerJoinNodeSize(tree) >
                     connectContext.getSessionVariable().getCboMaxReorderNodeUseExhaustive()) {
+                if (Utils.hasUnknownColumnsStats(tree) && !FeConstants.runningUnitTest) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java:134]: Add comment here ?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinReorderLeftDeep.java [hunk 2]
```diff
@@ -30,6 +30,7 @@ protected void enumerate() {
             ExpressionInfo joinExpr = buildJoinExpr(leftGroup, atoms.get(index));
             joinExpr.expr.deriveLogicalPropertyItself();
             calculateStatistics(joinExpr.expr);
+            computeCost(joinExpr, false);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinReorderLeftDeep.java:33]: Why not penalize cross join ?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java [hunk 3]
```diff
@@ -315,6 +319,21 @@ public static int countInnerJoinNodeSize(OptExpression root) {
...
+    public static boolean hasUnknownColumnsStats(OptExpression root) {
+        Operator operator = root.getOp();
+        if (operator instanceof LogicalScanOperator) {
+            LogicalScanOperator scanOperator = (LogicalScanOperator) operator;
+            List<String> colNames = scanOperator.getColumnRefMap().values().stream().map(Column::getName).collect(
+                    Collectors.toList());
+
+            List<ColumnStatistic> columnStatisticList =
+                    Catalog.getCurrentStatisticStorage().getColumnStatistics(scanOperator.getTable(), colNames);
+            return columnStatisticList.stream().anyMatch(c -> c.isUnknown() || c.isUnknownValue());
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java:331]: isUnknown() is enough here, isUnknownValue() is just used for plan replay

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java:136]: done

**Review by Youngwb — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by Seaven** [comment]
run starrocks_clang-format

**Comment by wanpengfei-git** [comment]
run starrocks_clang-format

**Comment by wanpengfei-git** [comment]
run starrocks_clang-format

**Pull request merged**